File: 프로그래머스/Level2/멀쩡한 사각형.cpp
using namespace std;

long long solution(int w,int h) {
    long long answer = 1;
    
    long long sum = (long long)w * (long long)h;
    
    int startPoint = w < h ? w : h;
    int k = 0;
    for(int i=startPoint; i>0; i--){
        if(w%i == 0 && h%i == 0){
            k = i;
            break;
        }
    }
    answer = sum - k*((w/k)+(h/k)-1);
    return answer;
}

File: 프로그래머스/Level2/멀쩡한 사각형.js
function solution(w, h) {
    const g = gcd(w,h); // 최대공약수

    return (w*h)-(w+h-g);
}

function gcd(a, b){
    const tmp = a%b;
    
    if(tmp === 0){
        return b;
    }
    else return gcd(b, tmp);
}
